lib/matplotlib/colors.py

```diff
@@ -789,7 +789,7 @@ def __init__(self, colors, name='from_list', N=None):
         if N is None:
             N = len(self.colors)
         else:
-            if (cbook.is_string_like(self.colors) and
+            if (isinstance(self.colors, six.string_types) and
                     cbook.is_hashable(self.colors)):
```

Comment:
probably was the case of masked string arrays.  removed the now redundant check.